src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioErrorReporterFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Shared]
+    [ExportWorkspaceServiceFactory(typeof(ErrorReporter), ServiceLayer.Host)]
+    internal class VisualStudioErrorReporterFactory : IWorkspaceServiceFactory
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        [ImportingConstructor]
+        public VisualStudioErrorReporterFactory(SVsServiceProvider serviceProvider)
```

Comment:
Do the `[Import(typeof(SvsServiceProvider))] IServiceProvider serviceProvider` to be explicit about our expectations. This is also consistent with what we do elsewhere.